runtime/lua/vim/_meta/lpeg.lua

```diff
@@ -0,0 +1,328 @@
+---@meta
+
+--These types were taken from https://github.com/LuaCATS/lpeg, with types being renamed to include
+--the vim namespace and with some descriptions made less verbose.
```

Comment:
AFAIK our convention is to put a space after the `--` in comments. Wonder if this can be checked by lint.

```suggestion
-- These types were taken from https://github.com/LuaCATS/lpeg, with types being renamed to include
-- the vim namespace and with some descriptions made less verbose.
```

Reply:
Is that convention only for `--` or also for `---`?

Reply:
Everywhere afaik. There is some old code where we omitted space after `---` because of limitations with the docs generator, but the generator no longer requires that.

Reply:
That's not quite right --- _text_ comments get the space, but annotations still need no space (so `---@param`, not `--- @param`).

Reply:
> but annotations still need no space (so `---@param`, not `--- @param`).

Oh I thought that was fixed

Reply:
No that's not correct. I've been adding spaces to annotations in new code.

Reply:
There doesn't seem to be a well-established convention though. The generator [always adds spaces](https://github.com/neovim/neovim/blob/d1bc6fca7123d77ebf03d0badd63f930a5b74064/runtime/lua/vim/_meta/api.lua), but in other places `---@` doesn't have a space (examples [here](https://github.com/neovim/neovim/blob/d1bc6fca7123d77ebf03d0badd63f930a5b74064/runtime/lua/vim/_meta/json.lua) and [here](https://github.com/neovim/neovim/blob/d1bc6fca7123d77ebf03d0badd63f930a5b74064/runtime/lua/vim/treesitter/dev.lua)).

Until we have an automated way to enforce whatever convention we want, I'll leave this as is.

Reply:
We're enforcing it now.

If you don't change it now someone else will need to fix it later, therefore making the process harder.

Reply:
Fine, I don't want this PR to get blocked by something like this, but these rules should be properly documented ([the docs say that `@---` shouldn't have a space afterwards](https://github.com/neovim/neovim/blob/d1bc6fca7123d77ebf03d0badd63f930a5b74064/CONTRIBUTING.md#lua-docstrings)) and preferably enforced by CI or some automated tool then.

Reply:
Yes more automation and documented policies are always welcome.

Reply:
As the one who claimed that this is the way, the onus is on you.